providerframework/metaservice/archiver/manager/manager_test.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2022 The SODA Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package manager
+
+import (
+	"fmt"
+	"os"
+	"testing"
+
+	"github.com/soda-cdm/kahu/providerframework/metaservice/archiver"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/suite"
+)
+
+type fakeStruct struct{}
+
+func (f *fakeStruct) Write(p []byte) (n int, err error) {
+	return 1, nil
+}
+
+func (f *fakeStruct) Close() error {
+	return nil
+}
+
+func (f *fakeStruct) Read(p []byte) (n int, err error) {
+	return 1, nil
+}
+
+type ManagerTestSuite struct {
+	suite.Suite
+	mgr          *archivalManager
+	archFileName string
+	typ          archiver.CompressionType
+	fwriter      *fakeStruct
+	fReader      *fakeStruct
+	invoke       func(Writer archiver.Writer) archiver.Writer
+	readInvoke   func(Reader archiver.Reader) archiver.Reader
+	archFilePath string
+}
+
+func (suite *ManagerTestSuite) BeforeTest(suiteName, testName string) {
+
+	switch testName {
+	case "TestGetArchiver":
+		suite.mgr = &archivalManager{
+			archiveYard: "fakeArchiveYard1",
+		}
+		suite.archFileName = "FakeArchfileName"
+
+		err := os.Mkdir("fakeArchiveYard1", 0777)
+		if err != nil {
+			fmt.Print(err)
+		}
+
+		suite.typ = archiver.CompressionType("fake")
+		suite.invoke = func(Writer archiver.Writer) archiver.Writer {
+			return suite.fwriter
+		}
+		RegisterCompressionWriterPlugins(suite.typ, suite.invoke)
+
+	case "TestGetArchiveReader":
+
+		suite.mgr = &archivalManager{
+			archiveYard: "fakeArchiveYard2",
+		}
+		err := os.Mkdir("fakeArchiveYard2", 0777)
+		if err != nil {
+			fmt.Print(err)
+		}
+		_, err = os.Create("fakeArchiveYard2/FakeArchfileName")
+		if err != nil {
+			fmt.Print(err)
+		}
+
+		suite.typ = archiver.CompressionType("name")
+
+		suite.fReader = &fakeStruct{}
+		suite.readInvoke = func(Reader archiver.Reader) archiver.Reader {
+			return suite.fReader
+		}
+		RegisterCompressionReaderPlugins(suite.typ, suite.readInvoke)
+
+		suite.archFilePath = "fakeArchiveYard2/FakeArchfileName"
+	}
```

Comment:
Please add a common function to initialise common utilities

Reply:
added  funcs to call instead